deploy/wmagent-venv-deploy.sh

```diff
@@ -0,0 +1,760 @@
+    do
+        vLen=${#var}
+        spaceLen=$(($valAllign - $vLen))
+        space=""
+        for ((i=0; i<=$spaceLen; i++))
+        do
+            space="$space "
+        done
+        spaceNewLineLen=$(($spaceLen + $prefixLen +$vLen))
+        spaceNewLine=""
+        for ((i=0; i<=$spaceNewLineLen; i++))
+        do
+            spaceNewLine="$spaceNewLine "
+        done
+        echo -e "WMCoreVenvVars[$var]${space}: ${WMCoreVenvVars[$var]//:/\n$spaceNewLine}"
+
+    done
+}
+
+wmaInstall() {
+    # The main function to setup/add the WMAgent virtual environment variables
+    # and call the install.sh script from: https://github.com/dmwm/CMSKubernetes/blob/master/docker/pypi/wmagent/install.sh
+
+    venvPath=$(_realPath $venvPath)
+    # TODO: To be changed to point to master at https://github.com/dmwm/CMSKubernetes.git
+    #       once we have it all merged at CMSKubernetes
+    deployRepo=https://github.com/todor-ivanov/CMSKubernetes.git
+    deployBranch=Fix_WMAgentPyPi_DcokerImage
+    echo "Cloning $deployRepo at $venvPath"
+    cd $venvPath
+    git clone $deployRepo
+    cd $venvPath/CMSKubernetes
+    git checkout $deployBranch
+    cd $venvPath/CMSKubernetes/docker/pypi/wmagent
+
+    export WMA_TAG=$wmTag
+    export WMA_USER=$(id -un)
+    export WMA_GROUP=$(id -gn)
+    export WMA_UID=$(id -u)
+    export WMA_GID=$(id -g)
+    export WMA_ROOT_DIR=$venvPath
+
+    # Basic WMAgent directory structure passed to all scripts through env variables:
+    # NOTE: Those should be static and depend only on $WMA_BASE_DIR
+    export WMA_BASE_DIR=$WMA_ROOT_DIR/srv
+    export WMA_ADMIN_DIR=$WMA_ROOT_DIR/admin/wmagent
+    export WMA_CERTS_DIR=$WMA_ROOT_DIR/certs
+
+    export WMA_HOSTADMIN_DIR=$WMA_ADMIN_DIR/hostadmin
+    export WMA_CURRENT_DIR=$WMA_BASE_DIR/wmagent/current
+    export WMA_INSTALL_DIR=$WMA_CURRENT_DIR/install
+    export WMA_CONFIG_DIR=$WMA_CURRENT_DIR/config
+    export WMA_MANAGE_DIR=$WMA_CONFIG_DIR/wmagent
+    export WMA_DEPLOY_DIR=$venvPath
+    export WMA_ENV_FILE=$WMA_DEPLOY_DIR/deploy/env.sh
+
+    # Setting up users and previleges
+    sudo groupadd -g ${WMA_GID} ${WMA_GROUP}
+    sudo useradd -u ${WMA_UID} -g ${WMA_GID} -m ${WMA_USER}
+    # sudo install -o ${WMA_USER} -g ${WMA_GID} -d ${WMA_ROOT_DIR}
+    sudo usermod -aG mysql ${WMA_USER}
+
+    # Add WMA_USER to sudoers
+    # sudo echo "${WMA_USER} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+
+    # Add all deployment needed directories
+    cp -rv bin/* $WMA_DEPLOY_DIR/bin/
+    cp -rv etc $WMA_DEPLOY_DIR/
+    cp -rv install.sh ${WMA_ROOT_DIR}/install.sh
+
+    # Add install script
+    cp -rv  install.sh ${WMA_ROOT_DIR}/install.sh
+
+    # Add wmagent run script
+    cp -rv run.sh ${WMA_ROOT_DIR}/run.sh
+
+    cd $WMA_ROOT_DIR
+
+    # Remove the already unneeded CMKubernetes repository:
+    rm -rf $venvPath/CMSKubernetes
+
+    # Install the requested WMA_TAG.
+    ${WMA_ROOT_DIR}/install.sh -v ${WMA_TAG}
```

Comment:
When we install WMAgent through this `install.sh` script, which under the hood will use `pip install`, it is supposed to bring in all of the python dependencies with the package.

This makes me wonder if there is any actual need for the function `setupDependencies`? I guess the answer is "it depends". If we want to have all of the WMCore dependencies, then the best is to install all those packages. But given that this script is being provided to the T0 to install WMAgent from PyPi, I would say that we should not install all of the WMCore dependencies, but only those that are actually needed by T0 agent.

Reply:
You are correct. 
It would be needed only for `run from source` mode. So if  in this mode we also deploy the service package, indeed it won't be needed, but if we don't, there is the need for it.

Reply:
fixed  now